zad2/warriors.rb

```diff
@@ -1,12 +1,113 @@
-class Warrior
+class Character
+  attr_accessor :name
+  attr_accessor :level
+  MIN_LEVEL = 1
+  MAX_LEVEL = 99
+
   def initialize(name:, level:)
+    raise ArgumentError.new('Level has to be in range <1:99>') if !checkLevel(level)
+    self.name = name
+    self.level = level
   end
 
   def strength
+    [*1..12].sample + self.level
+  end
+
+  def card
+    puts "#{self.name} (lvl #{self.level})"
+  end
+
+  def checkLevel(level)
+    level.between?(1,99)
   end
+
+
+end
+
+class Warrior < Character
+  def add_level(opponent)
+    if self.level <= opponent.level
+      self.level = checkLevel(self.level + (opponent.level - self.level + 1)) ? self.level + (opponent.level - self.level + 1) : 99
+      raise ArgumentError.new('Level is maximal!') if !checkLevel(self.level + (opponent.level - self.level))
+
+    else
+      self.level = checkLevel(self.level + 1) ? (self.level + 1) : self.level 
+      raise ArgumentError.new('Level is maximal!') if !checkLevel(self.level + 1)
+
+    end    
+  end
+end
+
+class Monster < Character
 end
 
 class BattleArena
+  attr_accessor :first_character, :second_character
+
   def initialize(first_character, second_character)
+    raise ArgumentError.new('Characters have to be Warriors or Monsters') if !checkCharacter(first_character, second_character)
+    self.first_character = first_character
+    self.second_character = second_character
   end
+
+  def checkCharacter(_first_character , _second_character)
```

Comment:
wg rubiowej konwencji - jeśli nazwa parametru zaczyna się od `_` to znaczy, że nie będzie używany